Add skip_download get parameter

davewalter opened the pull request:
This PR adds the ability to skip downloading the file on a get. Useful for skipping the implicit get after a put, or for triggering a job that doesn't actually use the file.

Regards,
Dave

davewalter committed:
Add skip_download get parameter

rizwanreza commented on README.md:
```diff
@@ -50,6 +50,10 @@ Places the following files in the destination:
+* `skip_download`: *Optional.* If true, skip downloading object from GCS.
+
+  This is useful to trigger a job that does not utilize the file, or to skip the implicit `get` after uploading a file to GCS using `put` (using `get_params`).
+
```
This is amazing! 🎊

syslxg commented:
We need this too.

syslxg commented:
in the s3 resource, the `skip_download` is a string instead of bool. do you want to keep consistent with s3 resource? @davewalter

davewalter commented:
Hi @syslxg,

I think `skip_download` is defined as a string in the `get` params because it also exists as a `source` configuration. It is defined as a `bool` in the source, but as a string in the `get` so that they can tell if it has been set there at all. Defining it as a bool in the `get` would default it to false, and then it wouldn't be possible to determine if this was intentional or just the default value.

I could add support for the `source` definition if you think it would be good to match the `s3-resource`. Let me know what you think.

Regards,
Dave

davewalter committed:
Update implementation of skip_download parameter

- flag can now be set in either the resource source definition, or the
get params
- matches the s3-resource

syslxg commented:
Now this works great in our pipelines. Thanks for making this PR, @davewalter

rizwanreza commented:
This would help with our work. 👍

frodenas commented:
Thanks @davewalter
